server/public/model/config.go

```diff
@@ -3456,18 +3456,26 @@ func (s *PluginSettings) Sanitize(pluginManifests []*Manifest) {
 		manifest := manifestMap[id]
 
 		for key := range settings {
-			if manifest == nil {
-				// Don't return plugin settings for plugins that are not installed
+			if manifest == nil ||
+				manifest.SettingsSchema == nil {
+				// Don't return any stored plugin settings if
+				//   - The plugin is no longer installed
+				//   - The plugin doesn't define any settings
 				delete(s.Plugins, id)
 				break
 			}
 
+			// notASecret is true when the plugin declared the settings key and it's not declared as secret
+			var notASecret bool
 			for _, definedSetting := range manifest.SettingsSchema.Settings {
-				if definedSetting.Secret && strings.EqualFold(definedSetting.Key, key) {
-					settings[key] = FakeSetting
+				if strings.EqualFold(definedSetting.Key, key) && !definedSetting.Secret {
+					notASecret = true
 					break
 				}
 			}
+			if !notASecret {
```

Comment:
Is the double negation truly necessary or could we simplify?

Reply:
I think the double negation is nedeed here. In case that the plugin doesn't definbe the setting any longer, we need to obfuscate it.